src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -1,23 +1,53 @@
 package core.basesyntax;
 
+//import java.util.Arrays;
+
 /**
  * Реалізуйте методи для знаходження головної та побічної діагоналі квадратної матриці,
  * максимального та мінімального елементів.
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        //matrix = new int [matrix.length][matrix.length];
+        int[] diagonalElts = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            diagonalElts[i] = matrix[i][i];
+        }
+        return diagonalElts;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] countDiagElts = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            countDiagElts[i] = matrix[i][matrix.length - (i + 1)];
```

Comment:
```suggestion
            counterDiagonalElts[i] = matrix[i][matrix.length - i + 1];
```
use full words at naming. it is much more understandable

Reply:
Pasha god morning. Comments deleted fm code.